extension_test.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2018 David Crawshaw <david@zentus.com>
+//
+// Permission to use, copy, modify, and distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+package sqlite_test
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"crawshaw.io/sqlite"
+)
+
+const (
+	extCode = `
+/*
+** Copyright (c) 2018 David Crawshaw <david@zentus.com>
+**
+** Permission to use, copy, modify, and distribute this software for any
+** purpose with or without fee is hereby granted, provided that the above
+** copyright notice and this permission notice appear in all copies.
+**
+** THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+** WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+** MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+** ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+** WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+** ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+** OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+
+#include "sqlite3ext.h"
+SQLITE_EXTENSION_INIT1
+
+#include <stdlib.h>
+
+static void hellofunc(
+	sqlite3_context *context,
+	int argc,
+	sqlite3_value **argv){
+		(void)argc;
+		(void)argv;
+		sqlite3_result_text(context, "Hello, World!", -1, SQLITE_STATIC);
+	}
+
+#ifdef _WIN32
+__declspec(dllexport)
+#endif
+int sqlite3_hello_init(
+	sqlite3 *db,
+	char **pzErrMsg,
+	const sqlite3_api_routines *pApi
+){
+		int rc = SQLITE_OK;
+		SQLITE_EXTENSION_INIT2(pApi);
+		(void)pzErrMsg;  /* Unused parameter */
+		return sqlite3_create_function(db, "hello", 0, SQLITE_UTF8, 0, hellofunc, 0, 0);
+}`
+)
+
+func TestLoadExtension(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "sqlite")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpdir)
+	fout, err := os.Create(filepath.Join(tmpdir, "ext.c"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	io.Copy(fout, strings.NewReader(extCode))
+	fout.Close()
```

Comment:
as you are writing a file, check the error here and on the copy.

as that gets tedious, use ioutile:

if err := ioutile.WriteFile(file path.Join(tmpdir, "ext.c")), []byte(extCode), 0666); err != nil {
...
}

Reply:
Understood.